### PR TITLE
Fix flake.lock deserialization

### LINKED ISSUE #34: error: the string is not allowed to refer to a store path.
I get the same error message whether I `nix run` or `nix profile install ...`:

```
$ nix run github:DeterminateSystems/flake-checker
error: the string '# This file is automatically @generated by Cargo.
       # It is not intended for manual editing.
       version = 3

       [[package]]
       name = "addr2line"
       version = "0.19.0"
       source = "registry+https://github.com/rust-lang/crates.io-index"
       checksum = "a76fd60b23679b7d19bd066031410fb7e458ccc5e958eb5c325888ce4baedc97"
       dependencies = [
        "compiler_builtins",
        "gimli 0.27.2",
        "rustc-std-workspace-alloc",
        "rustc-std-workspace-core",
       ]

       [[package]]
...
       [[package]]
       name = "zerovec-derive"
       version = "0.9.3"
       source = "registry+https://github.com/rust-lang/crates.io-index"
       checksum = "2154cb6e2a748163354165e22c6a555effb09ca2d16334767bf66bb404f2206e"
       dependencies = [
        "proc-macro2",
        "quote",
        "syn 1.0.102",
        "synstructure 0.12.6",
       ]
       ' is not allowed to refer to a store path (such as '')
```

```
$ nix --version
nix (Nix) 2.10.3

$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.2 LTS
Release:	22.04
Codename:	jammy
```

### PR DESCRIPTION
**Pull request by lucperkins**
Fixes #34

### CODE REVIEW

File: src/flake.rs [hunk 1]
```diff
@@ -178,9 +186,15 @@ impl<'de> Deserialize<'de> for FlakeLock {
...
-                    root_nodes.insert(root_name.to_owned(), root_node);
+                    let node_keys: Vec<String> = match root_reference {
+                        Input::String(s) => vec![s.to_string()],
+                        Input::List(keys) => keys.to_owned(),
+                    };
+
+                    for key in node_keys {
+                        let root_node = nodes[&key].to_owned();
+                        root_nodes.insert(root_name.to_owned(), root_node);
+                    }
```
Comment (neilmayhew) [inline src/flake.rs:197]: I don't think you've interpreted the meaning of the multiple keys correctly. It's a hierarchical list of keys, not a flat one. The flake format documentation in `nix flake --help` says,

> It is also possible to "inherit" an input from another input. This is useful to minimize flake
> dependencies. For example, the following sets the nixpkgs input of the top-level flake to be
> equal to the nixpkgs input of the dwarffs input of the top-level flake:
>
>   | inputs.nixpkgs.follows = "dwarffs/nixpkgs";
>
> The value of the follows attribute is a /-separated sequence of input names denoting the path of
> inputs to be followed from the root flake.

Comment (neilmayhew) [reply src/flake.rs:197]: The example we've been working with has this in the flake:

```nix
inputs.nixpkgs.follows = "haskellNix/nixpkgs-unstable";
```

which produces this in the lock:

```json
{
  "inputs": {
    "nixpkgs": [
      "haskellNix",
      "nixpkgs-unstable"
    ],
  }
}
```

Comment (lucperkins) [reply src/flake.rs:197]: I'm currently in the process of researching this and I have this PR marked as a draft for precisely that reason.

File: src/flake.rs [hunk 2]
```diff
@@ -202,18 +201,64 @@ impl<'de> Deserialize<'de> for FlakeLock {
...
+                    "lock node should have had some inputs but had none:\n{:?}",
+                    node
+                )));
+            }
+        };
+
+        let next_inputs = &node_inputs[&input];
+        node = match next_inputs {
+            Input::String(s) => &nodes[s],
+            Input::List(inputs) => chase_input_node(nodes, inputs.to_owned().into())?,
```
Comment (lucperkins) [inline src/flake.rs:238]: Go-go gadget recursion!

**Comment by lucperkins** [comment]
LGTM!

**Pull request merged**